src/Compilers/CSharp/Portable/Binder/SwitchExpressionArmBinder.cs

```diff
@@ -39,7 +39,7 @@ internal override BoundSwitchExpressionArm BindSwitchExpressionArm(SwitchExpress
             Binder armBinder = this.GetRequiredBinder(node);
             bool hasErrors = switchGoverningType.IsErrorType();
             ImmutableArray<LocalSymbol> locals = _armScopeBinder.Locals;
-            BoundPattern pattern = armBinder.BindPattern(node.Pattern, switchGoverningType, switchGoverningValEscape, permitDesignations: true, hasErrors, diagnostics);
```

Comment:
>switchGoverningValEscape [](http://example.com/codeflow?start=92&length=24)

The method signature for `BindSwitchExpressionArm` can be simplified and the caller too (no longer need to compute val escape)